### PR TITLE
Minor ClusterSharding scaladoc suggestions.

### PR DESCRIPTION
**Pull request by helena**
Readability - organization, clarity and grammar improvements.

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM, would be good if those headers showed up like actual headers in the docs though, make them bold perhaps?

**Comment by helena** [comment]
@johanandren good idea, updated with the  '''Some Header''' and Scaladoc lists.

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala [hunk 1]
```diff
@@ -41,81 +41,90 @@ import akka.util.ByteString
...
+ * knows the shard mappings and routes inbound messages to the entity with the entity id.
+ * Messages to the entities are always sent via the local `ShardRegion`.
  * The `ShardRegion` actor is started on each node in the cluster, or group of nodes
  * tagged with a specific role. The `ShardRegion` is created with two application specific
  * functions to extract the entity identifier and the shard identifier from incoming messages.
+ *
+ * Typical usage of this extension:
+ *   1. At system startup on each cluster node by registering the supported entity types with
+ * the [[ClusterSharding#start]] method
+ *   2. Retrieve the `ShardRegion` actor for a named entity type with [[ClusterSharding#shardRegion]]
```
Comment (raboof) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: Hmm this doesn't seem to be picked up correctly - '1' and '2' are one the same line for me 
DeepinScreenshot_select-area_20190327164412

**Review by raboof — COMMENTED** [review]
Great to split this up a bit and add headings - quite a wall of text still! One suggestion wrt the numbered list.

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala (hunk 1 above)
Comment (helena) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: Mysterious. I literally grabbed it from the Scaladoc guide with the spacing. Thanks so much for the catch.

Comment (helena) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala]: OIC what I did wrong. Pushing fix.

**Comment by helena** [comment]
PLS TEST

**Comment by helena** [comment]
PLS BUILD

**Pull request merged**